source/code/plugins/in_dsc_monitor.rb

```diff
@@ -85,6 +86,7 @@ def run_check_status
           @dsc_cache[:status] = dsc_status
           @dsc_cache.commit
         }
+        File.chmod(0644, dsc_cache_file)
```

Comment:
I'm surprised the YMAL::Store and its superclass don't seem to support providing the initial file permissions for when it needs to create the file.

Recommendation: Don't let the YAML::Store.new create the file. Always make sure it exists and is created with the minimal correct permissions **before** calling new. This ensure there is never a window time between creating the file and setting the permissions where it is too open.